vercel-api/src/components/charts/ChartBase.ts

```
@@ -0,0 +1,510 @@
+      const date = new Date(event.timestamp);
+      let key: string;
+      
+      switch (groupBy) {
+        case 'hour':
+          key = date.toISOString().substr(0, 13) + ':00:00Z';
+          break;
+        case 'day':
+          key = date.toISOString().substr(0, 10) + 'T00:00:00Z';
+          break;
+        case 'week':
+          const weekStart = new Date(date.setDate(date.getDate() - date.getDay()));
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)